src/CodeStyle/Core/CodeFixes/AbstractFormattingAnalyzerImpl.cs

```diff
@@ -0,0 +1,94 @@
+
+            var options = workspace.Options;
+            if (File.Exists(tree.FilePath))
+            {
+                var codingConventionContext = codingConventionsManager.GetConventionContextAsync(tree.FilePath, cancellationToken).GetAwaiter().GetResult();
+                options = ApplyFormattingOptions(options, codingConventionContext);
+            }
+
+            var oldText = tree.GetText(cancellationToken);
+            var formattingChanges = Formatter.GetFormattedTextChanges(tree.GetRoot(cancellationToken), workspace, options, cancellationToken);
+
+            // formattingChanges could include changes that impact a larger section of the original document than
+            // necessary. Before reporting diagnostics, process the changes to minimize the span of individual
+            // diagnostics.
+            foreach (var formattingChange in formattingChanges)
+            {
+                var change = formattingChange;
+                if (change.NewText.Length > 0 && !change.Span.IsEmpty)
+                {
+                    // Handle cases where the change is a substring removal from the beginning. In these cases, we want
+                    // the diagnostic span to cover the unwanted leading characters (which should be removed), and
+                    // nothing more.
+                    var offset = change.Span.Length - change.NewText.Length;
+                    if (offset >= 0)
+                    {
+                        if (oldText.GetSubText(new TextSpan(change.Span.Start + offset, change.NewText.Length)).ContentEquals(SourceText.From(change.NewText)))
+                        {
+                            change = new TextChange(new TextSpan(change.Span.Start, offset), "");
+                        }
+                        else
+                        {
+                            // Handle cases where the change is a substring removal from the end. In these cases, we want
+                            // the diagnostic span to cover the unwanted trailing characters (which should be removed), and
+                            // nothing more.
+                            if (oldText.GetSubText(new TextSpan(change.Span.Start, change.NewText.Length)).ContentEquals(SourceText.From(change.NewText)))
+                            {
+                                change = new TextChange(new TextSpan(change.Span.Start + change.NewText.Length, offset), "");
+                            }
+                        }
+                    }
+                }
+
+                if (change.NewText.Length == 0 && change.Span.IsEmpty)
+                {
+                    // No actual change
+                    throw new InvalidOperationException("This program location is thought to be unreachable.");
+                }
+
+                var location = Location.Create(tree, change.Span);
+                context.ReportDiagnostic(DiagnosticHelper.Create(
```

Comment:
>DiagnosticHelper.Create [](start = 41, length = 23)

Why not `Diagnostic.Create`? The severity of the diagnostic seems to be unaffected by options.